data/src/server.rs

```diff
@@ -20,19 +21,107 @@ impl AsRef<str> for Server {
+    pub ping_time: Option<u32>,
+    /// The amount of time in seconds for a client to reconnect due to no ping response.
+    pub ping_timeout: Option<u32>,
+    /// The length in seconds of a rolling window for message throttling. If more than
+    /// `max_messages_in_burst` messages are sent within `burst_window_length` seconds, additional
+    /// messages will be delayed automatically as appropriate. In particular, in the past
+    /// `burst_window_length` seconds, there will never be more than `max_messages_in_burst` messages
+    /// sent.
+    pub burst_window_length: Option<u32>,
+    /// The maximum number of messages that can be sent in a burst window before they'll be delayed.
+    /// Messages are automatically delayed as appropriate.
+    pub max_messages_in_burst: Option<u32>,
+    /// Whether the client should use NickServ GHOST to reclaim its primary nickname if it is in
+    /// use. This has no effect if `nick_password` is not set.
+    #[serde(default)]
+    pub should_ghost: bool,
+    /// The command(s) that should be sent to NickServ to recover a nickname. The nickname and
+    /// password will be appended in that order after the command.
+    /// E.g. `["RECOVER", "RELEASE"]` means `RECOVER nick pass` and `RELEASE nick pass` will be sent
+    /// in that order.
+    pub ghost_sequence: Option<Vec<String>>,
+    /// A mapping of channel names to keys for join-on-connect.
+    #[serde(default)]
+    pub channel_keys: HashMap<String, String>,
 }
 
-impl Deref for Config {
-    type Target = irc::client::data::Config;
-
-    fn deref(&self) -> &Self::Target {
-        &self.0
+impl From<Config> for irc::client::data::Config {
+    fn from(config: Config) -> Self {
+        irc::client::data::Config {
+            nickname: config.nickname,
+            nick_password: config.nick_password,
+            alt_nicks: config.alt_nicks,
+            username: config.username,
+            realname: config.realname,
+            server: config.server,
+            port: config.port,
+            password: config.password,
+            use_tls: config.use_tls,
+            cert_path: config.cert_path,
+            client_cert_path: config.client_cert_path,
+            client_cert_pass: config.client_cert_pass,
+            dangerously_accept_invalid_certs: config.dangerously_accept_invalid_certs,
+            encoding: config.encoding,
+            channels: config.channels,
+            umodes: config.umodes,
+            user_info: config.user_info,
+            ping_time: config.ping_time,
+            ping_timeout: config.ping_timeout,
+            burst_window_length: config.burst_window_length,
+            max_messages_in_burst: config.max_messages_in_burst,
+            should_ghost: config.should_ghost,
+            ghost_sequence: config.ghost_sequence,
+            channel_keys: config.channel_keys,
+            version: Some(format!("Halloy {VERSION}")),
+            owners: vec![],
+            source: None,
```

Comment:
Not entirely sure how to do a `CTCP SOURCE` request, and what it should display.